src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/UpgradeInformationEvent.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Common;
+using NuGet.ProjectManagement;
+
+namespace NuGet.PackageManagement.Telemetry
+{
+    public class UpgradeInformationTelemetryEvent : TelemetryEvent
+    {
+        public UpgradeInformationTelemetryEvent()
+            : base("UpgradeInformation")
+        {
+        }
+
+        public void SetResult(IEnumerable<NuGetProject> projects, NuGetOperationStatus status, int packageCount)
+        {
+            var sortedProjects = projects.OrderBy(
+                project => project.GetMetadata<string>(NuGetProjectMetadataKeys.UniqueName));
+
+            var projectIds = sortedProjects.Select(
+                project => project.GetMetadata<string>(NuGetProjectMetadataKeys.ProjectId)).ToArray();
```

Comment:
I think we can avoid calling `.ToArray()` because [`String.Join`](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netcore-3.1#System_String_Join__1_System_String_System_Collections_Generic_IEnumerable___0__) and [`StringBuilder.AppendJoin`](https://docs.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.appendjoin?view=netcore-3.1#System_Text_StringBuilder_AppendJoin__1_System_String_System_Collections_Generic_IEnumerable___0__) have an overloaded method that can accept `IEnumerable` as parameter. Thoughts ?

Reply:
I think so...